OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -64,7 +64,8 @@ public PathFinder(World world)
 
 		public List<CPos> FindUnitPath(CPos source, CPos target, Actor self, Actor ignoreActor, BlockedByActor check)
 		{
-			var mobile = self.Trait<Mobile>();
+			// PERF: Because we can be sure that OccupiesSpace is Mobile here, we can save some performance by avoiding querying for the trait.
+			var mobile = self.OccupiesSpace as Mobile;
```

Comment:
It is more efficient to cast directly, like `var locomotor = ((Mobile)self.OccupliesSpace).Locomotor`.